microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -168,6 +169,37 @@ def _check_rx_or_tx_pps(
                 "than 2^20 (~1m) PPS after sriov disable."
             ).is_less_than(2 ** 20)
 
+    @TestCaseMetadata(
+        description="""
+            verify vpp is able to detect azure network interfaces
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_vpp(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+
+        vpp = DpdkVpp(node)
+        vpp.install()
+        vpp.install_from_package_manager()
+
+        net = Nics(node)
+        net.initialize()
+        _, nic = net.get_test_nic()
+
+        # set devices to down and restart vpp service
+        for dev in [nic.lower, nic.upper]:
+            node.execute(f"ip link set {dev} down", sudo=True, expected_exit_code=0)
+        for dev in [nic.lower, nic.upper]:
+            node.execute(f"ip addr flush dev {dev}", sudo=True, expected_exit_code=0)
```

Comment:
Add the implementation into the ip.py.

Reply:
https://github.com/microsoft/lisa/pull/1686